app/controllers/concerns/pagination.rb

```diff
@@ -3,9 +3,12 @@
 # Default options and configuration for pagination
 module Pagination
   extend ActiveSupport::Concern
+
+  ParamType = ActionController::Parameters
+
   included do
     def pagination_limit
-      params[:limit].present? ? params[:limit].to_i : 10
+      params.permit(limit: ParamType.integer & ParamType.gt(0))[:limit] || 10
```

Comment:
You can slice (select) only the limit to prevent `Unpermitted parameter` on extra parameters:
```
params.slice(:limit).permit(...
```

Reply:
I do not recommend this. The unpermitted params log only happens in development and testing environments and is just a warning. Once we properly validate _all_ our params, the warnings will go away.

Reply:
Scratch that:

> By default parameter keys that are not explicitly permitted will be logged in the development and test environment. In other environments these parameters will simply be filtered out and ignored.

We definitely don't want them filtered out and ignored :laughing:

Reply:
I hope that they are filtered out only from the return value of `permit` and not from whole `params`. A slice would not hurt it here, as it creates a copy to leverage only one parameter validation.

Reply:
I tried testing with RAILS_ENV=production, and it seems to be OK. The slice also seems fine.